src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private double firstLeg;
+    private double secondLeg;
+
+    public RightTriangle(Color color, double firstLeg, double secondLeg) {
+        super(color);
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+    }
+
+    @Override
+    public double calculateArea() {
+        return 0.5 * firstLeg * secondLeg;
+    }
+
+    @Override
+    public void drawFigure() {
```

Comment:
Don't put all behavior into a single interface if the methods are conceptually different from each other. Here, the `drawFigure()` method should not be in the same interface as `calculateArea()` since drawing and calculating area are conceptually different. Consider separating these concerns into different interfaces.